inc/zoo/swar/SWAR.h

```diff
@@ -62,6 +63,13 @@ constexpr std::make_unsigned_t<T> lsbIndex(T v) noexcept {
 template<int NBits_, typename T = uint64_t>
 struct SWAR {
     using type = std::make_unsigned_t<T>;
+    constexpr static auto calculateLeastSignificantLaneMask() {
+      const auto m =
+            sizeof(T) * 8 == NBits ? // needed to avoid shifting all bits
+                ~T(0) :
+                ~(~T(0) << NBits);
+      return m;
+    }
     constexpr static inline type
```

Comment:
Nah...
This calculation does not deserve a name.
I think since C++17 we can use constexpr lambdas, that would be more appropriate below.
However, I showed you that this calculation is not needed, you're repating, and that's the bad part, the bulk of the work done in other constants.
Better than repeating the work is to use the work already done, primarily because then you have cross-checking that things work, if there's a mistake in the old, the new might discover it, if further nuance is discovered the new might benefit from it directly.

Reply:
Thanks, should have reverted. This was a debugging attempt. Was getting a red herring compiler message about this variable not being constexpr when trying to implement `to_array`.